### ISSUE TITLE
iOS: Live reload seems broken (Mac OS)

### ISSUE THREAD

**Issue by henninghorn** [centrual/cordova-template-framework7-vue-webpack#1]
Hi @centrual 

Somehow the live reload functionality seems broken for me, when I run:
$ `cordova run ios -- --live-reload` from my project folder

It's a completely clean project.

I get the following from the iOS simulator:
![skaermbillede 2017-01-14 kl 12 09 10](https://cloud.githubusercontent.com/assets/742607/21954491/5dad1570-da52-11e6-9722-5fd59f7507a6.png)

I'm running Cordova 6.4.0 and Node 7.2.1.

I'm not sure whether the template is misconfigured, or the cordova browsersync plugin is broken:
- http://stackoverflow.com/questions/35975086/why-is-the-cordova-browser-sync-plugin-not-working-on-a-clean-new-apache-cordov

Do the live reload feature work on your machine?

**Comment by centrual** [comment]
Same page coming to me on my computer too, but in my computer, after i pressed button on top, page coming. Also in my android phone, it redirects automatically. Can you try buttons for check?

cordova-plugin-browsersync not related with me, i'm using it for development. but i can make new version of it if it's not working for you. 😄 

**Update:** I started to make new cordova browsersync plugin. Because other plugin not maintaining. Also i don't like the other plugin, it's slow, installs node_modules multiple times and can't handle some things. I think i'll complete in 3 days.

**Comment by henninghorn** [comment]
$ `cordova emulate browser -- --live-reload`
This gives me the browsersync error screen when it starts, but after clicking the link I get into the app. And visiting the `localhost:3000` URL manually also works great.

$ `cordova emulate ios -- --live-reload`
This also gives me the browsersync error screen, but clicking the links have no effect.
Manually visiting the URL doesn't work as well, the browsersync server doesn't seem to be running at all.

So my guess is that somehow the browsersync server doesn't start or it somehow gets shutdown, when you're dealing with the iOS platform - unfortunately.

Regarding the new plugin, that sounds great, @centrual! Live reload is really great. When building Vue projects I'm using the official Webpack template which comes with something called hot-reload, which keep the current state of your app, and only components you edit gets refreshed.

I'm looking forward to test your plugin 👍

**Comment by centrual** [comment]
I searched about localhost connection in ios simulator. I think there is a bug about localhost connection inside ios simulator. Are you using latest version of xCode and ios?

I checked the solution and i found this changelog:

https://developer.apple.com/library/content/documentation/Xcode/Conceptual/RN-Xcode-Archive/Chapters/xc7_release_notes.html#//apple_ref/doc/uid/TP40016994-CH5-SW2

They saying:
```
When running in an iOS simulator for iOS versions 9.1 and earlier, an app may not communicate
with TCP/IP services hosted locally by the Mac using the Mac’s local IP address.
Connect via a loopback address, such as localhost or 127.0.0.1. (22453539)
```

Your idea about webpack-dev-server is very good for this project, i think, i don't need browsersync aynmore. Let i use webpack-dev-server with hot-reload for this project.

**Comment by henninghorn** [comment]
I've tried opening `localhost:3000` on my Mac which is running the iOS simulator, and the browsersync server doesn't seem to run at all.

But when running `cordova emulate browser -- --live-reload` the server is up and running.

Btw I was running iOS 10.2 on the simulator :)

**Comment by centrual** [comment]
ooh, i understand better now. I am sorry, my English not fluent, i trying to learn... 👼

I'll fix this problem on new webpack-dev-server version.

**Comment by henninghorn** [comment]
No worries 😄

**Comment by centrual** [comment]
I made everything. ~everything working except hot-module-replacement. When i change something, page becomes white screen and there is no error.~

~Is anybody knows how can i configure hot-module-replacement?~

I added hot module replacement too! 🙌

**Comment by centrual** [comment]
Can you check with latest release @henninghorn . It must be fixed.

**Comment by henninghorn** [comment]
Hi @centrual 

I've just tried, but unfortunately I'm getting the following error when I make a change to a file while running in live reload mode:

````
logPath: /Users/henning/Code/framework7/cordova/HelloWorld/platforms/ios/cordova/console.log
 ~/Code/framework7/cordova/HelloWorld  events.js:160
      throw er; // Unhandled 'error' event
      ^

Error: write EPIPE
    at exports._errnoException (util.js:1022:11)
    at WriteWrap.afterWrite [as oncomplete] (net.js:804:14)
````

If I visit `localhost:8080` from my browser on my computer I see the F7 app, but as soon as I make a change the Webpack server crashes and I get the error above.

I do not now what is causing this.

If I run `cordova emulate browser -- --live-reload` everything works fine, reload works without any problems.

**Comment by centrual** [comment]
@henninghorn Thanks for feedback. I haven't mac and iPhone, but i'll try inside virtual machine in 1 week.

(Note: I think EPIPE can be related with write a file. Can you add privilenge to main folder for write and read)

Have a great day!

**Comment by centrual** [comment]
@henninghorn I updated webpack-dev-server and webpack to 2.2.0 Release. Can you check with new webpack release?

**Comment by henninghorn** [comment]
@centrual I just tried re-downloading your template, also ensuring that the template was indeed using webpack 2.2.0, but unfortunately it doesn't work.

````
logPath: /Users/henning/Code/framework7/cordova/HelloWorld/platforms/ios/cordova/console.log
 ~/Code/framework7/cordova/HelloWorld  events.js:160
      throw er; // Unhandled 'error' event
      ^

Error: write EPIPE
    at exports._errnoException (util.js:1022:11)
    at WriteWrap.afterWrite [as oncomplete] (net.js:804:14)
````

I don't believe the EPIPE error has something to do with permissions, I think it has something to do when "privilige" to output to the screen or something like that.

Normally when I run webpack in a Vue project, Webpack stays in the terminal, so it can tell me what it is doing. But when running `cordova emulate ios -- --live-reload` Webpack seems to get "pushed" to the background and because of that the error occurs when you edit a file and Webpacks wants to tell the terminal about it.
Does that make sense?

In other words, I believe something between these lines (183-216) might have something to do with it: https://github.com/centrual/cordova-template-framework7-vue-webpack/blob/master/template_src/hooks/hookers.js#L183-L216

**Comment by centrual** [comment]
Hello @henninghorn;

I installed mac os into the wmware. Checked the problems and i understand why it's happening.

Some problems in Mac OS:
1) Needs xCode and sometimes returns: `Error: xcode-select: error: tool 'xcodebuild' requires Xcode, but active developer directory is a command line tools instance`. This is the solution for error: https://github.com/nodejs/node-gyp/issues/569#issuecomment-255589932
2) 8080 port using in default mac os, so different port required for mac. (I changing the take port strategy, it will find free port automatically. )
3) Webpack's extract-text-plugin not working at all, it can't build webpack. ( I'm waiting for new release )
4) Build errors sometimes returns big error texts, but exec command's buffer is too small for handle it. ( Fixed )
5) On live reload mode, terminal closes the stream after launch app. So when we change some file in src folder, it handles the change but it can't write to console because of stdout stream is closed. So it returns `Error: write EPIPE` ( I'm looking fix for that )

I'll update the package after [extract-text-plugin](https://github.com/webpack/extract-text-webpack-plugin) update.

Have a great day!

**Comment by centrual** [comment]
Hello @henninghorn 

I updated template and readme file. I tested in MacOS. Can you read instructions from https://github.com/centrual/cordova-template-framework7-vue-webpack/ and test it?

Thank you so much.
Have a great day!

**Comment by henninghorn** [comment]
Hi @centrual 

Hot reload/live reload works by following the "WARNING"-instructions.

Great work! 👍

**Comment by centrual** [comment]
@henninghorn Thanks for feedback, i closing the issue.
